src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/generate_proposals.cpp

```diff
@@ -0,0 +1,197 @@
+                    0.30608943, 0.6572437,  0.69185436, 0.88646156, 0.36985755, 0.5590753,
+                    0.5256446,  0.03342898, 0.1344396,  0.68642473, 0.37953874, 0.32575172,
+                    0.21108444, 0.5661886,  0.45378175, 0.62126315, 0.26799858, 0.37272978,
+                    0.56637216, 0.90457034, 0.69827306, 0.4353543,  0.47985056, 0.42658508,
+                    0.14516132, 0.08081771, 0.1799732,  0.9229515,  0.42420176, 0.50857586,
+                    0.82664067, 0.4972319,  0.3752427,  0.56731623, 0.18241242, 0.33252355,
+                    0.30608943, 0.6572437,  0.69185436, 0.88646156, 0.36985755, 0.5590753,
+                    0.5256446,  0.03342898, 0.1344396,  0.68642473, 0.37953874, 0.32575172,
+                    0.21108444, 0.5661886,  0.45378175, 0.62126315, 0.26799858, 0.37272978}),
+            }
+        }
+    };
+    return input_tensors;
+}
+
+constexpr size_t num_batches = 2;
+constexpr size_t height = 2;
+constexpr size_t width = 6;
+constexpr size_t number_of_anchors = 3;
+
+const std::vector<std::vector<InputShape>> input_shape = {
+        // im_info / anchors / boxesdeltas / scores
+        static_shapes_to_test_representation({{num_batches, 3},
+                                              {height, width, number_of_anchors, 4},
+                                              {num_batches, number_of_anchors * 4, height, width},
+                                              {num_batches, number_of_anchors, height, width}}),
+};
+
+
+
+INSTANTIATE_TEST_SUITE_P(
+        smoke_GenerateProposalsLayerTest_f16,
+        GenerateProposalsLayerTest,
+        ::testing::Combine(
+                ::testing::ValuesIn(input_shape),
+                ::testing::ValuesIn(min_size),
+                ::testing::ValuesIn(nms_threshold),
+                ::testing::ValuesIn(post_nms_count),
+                ::testing::ValuesIn(pre_nms_count),
+                ::testing::ValuesIn({true}),
+                ::testing::ValuesIn(getInputTensors<ov::float16>()),
+                ::testing::ValuesIn({ov::element::Type_t::f16}),
+                ::testing::ValuesIn({ov::element::Type_t::i64}),
+                ::testing::Values(CommonTestUtils::DEVICE_GPU)),
+        GenerateProposalsLayerTest::getTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(
+        smoke_GenerateProposalsLayerTest_f32,
+        GenerateProposalsLayerTest,
+        ::testing::Combine(
+                ::testing::ValuesIn(input_shape),
+                ::testing::ValuesIn(min_size),
+                ::testing::ValuesIn(nms_threshold),
+                ::testing::ValuesIn(post_nms_count),
+                ::testing::ValuesIn(pre_nms_count),
+                ::testing::ValuesIn({false}),
+                ::testing::ValuesIn(getInputTensors<float>()),
+                ::testing::ValuesIn({ov::element::Type_t::f32}),
+                ::testing::ValuesIn({ov::element::Type_t::i32}),
```

Comment:
Please add test cases with combination of input precision(f16,f32) and roi_num_type(i32,i64) as unit test.

Reply:
They are already there in unittests - see testcases f32_i32, f32_i64, f16_i32, f16_i64 in
[generate_proposals_gpu_test.cpp](https://github.com/openvinotoolkit/openvino/pull/11994/files#diff-93586bb33b4c34544ed7eaff84982e48f05f1ffd2c7f28b4b3903c1f1695d4aeR418)

Reply:
Yes, we need to have those test cases in both unit test and functional single layer test.

Reply:
Added to single layer tests as well.